### ISSUE TITLE
Testinfra breaks parameterized test discovery

### ISSUE THREAD

**Issue by smarlowucf** [philpep/testinfra#232]
If a test is parameterized the testinfra plugin prepends the backend to test id's:

````
<Module 'test_stuff.py'>
  <Function 'test_stuff[local-test1]'>
  <Function 'test_stuff[local-test2]'>
  <Function 'test_stuff[local-test3]'>
````

Pytest only, yields:

````
<Module 'test_stuff.py'>
  <Function 'test_stuff[test1]'>
  <Function 'test_stuff[test2]'>
  <Function 'test_stuff[test3]'>
````

This makes it confusing to run only a specific parameter especially with a remote host.

No longer works:

`testinfra ... test_stuff[test1]`

Using --collect-only with no other options/args yields "local-" prepended to all parameterized id's. This is okay if you are running a test locally. However, if you run `--collect-only` to get a list of tests, then attempt to run against a paramiko host, the test does not exist.

`no name test_stuff[local-test1]`

Because the backend is now paramiko://{ip-address}. (test_stuff[paramiko://{ip_address}-test1]).

Is there a way to disable this if one or fewer hosts are supplied?

**Comment by philpep** [comment]
Yes, I think adding this information is useful only if there is multiple hosts supplied.

**Comment by smarlowucf** [comment]
@philpep I managed to disable it in [pytest_generate_tests](https://github.com/philpep/testinfra/blob/b8ae9c3c3963b654a4549b6b245d6e9ac4627fdb/testinfra/plugin.py#L128). However, that triggered other exceptions. Would this be the correct spot to make the patch?

**Comment by philpep** [comment]
@smarlowucf this is probably here, when ``params`` contains only one element there is only one host and you can set ``ids`` to ``[None]`` or ``['']`` or don't pass it to ``parametrize()``
